lib/modules.nix

```
@@ -115,8 +115,21 @@ rec {
 
       checkUnmatched =
         if config._module.check && config._module.freeformType == null && merged.unmatchedDefns != [] then
-          let inherit (head merged.unmatchedDefns) file prefix;
-          in throw "The option `${showOption prefix}' defined in `${file}' does not exist."
+          let
+            inherit (head merged.unmatchedDefns) file prefix;
+            filteredSet = builtins.removeAttrs (attrByPath (sublist 0 (length prefix - 1) prefix) {} options)
+              [ "_module" ];
+            baseMsg = "The option `${showOption (prefix)}' defined in `${file}' does not exist.";
+          in
+            if filteredSet == {}
```

After testing this a bit, I'm noticing that the error can trigger in other scenarios too, e.g. in a standard NixOS module (not a submodule) that looks like
```nix
{
  options.foo = {};
  config.foo.bar = {};
}
```

In such a case, seeing the error might be more confusing.

So instead I suggest to change this condition to just check whether there are _any_ options at all:
```suggestion
            if attrNames options == [ "_module" ]
```

This still triggers if no options are defined in the `submodule`, but it won't trigger in the above example.

---

Good catch!